File: src/main/java/musiccube/entities/ArtistInstrument.java
package musiccube.entities;

import javax.persistence.*;

@Entity
@Table(name = "artist_instrument")
public class ArtistInstrument {

    @Id
    @GeneratedValue
    @Column
    private int id;

    @ManyToOne
    @JoinColumn(name = "artist_id")
    private Artist artist;

    @ManyToOne
    @JoinColumn(name = "instrument_id")
    private Instrument instrument;

    public ArtistInstrument() {}

    public ArtistInstrument(Artist artist, Instrument instrument) {
        this.artist = artist;
        this.instrument = instrument;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Artist getArtist() {
        return artist;
    }

    public void setArtist(Artist artist) {
        this.artist = artist;
    }

    public Instrument getInstrument() {
        return instrument;
    }

    public void setInstrument(Instrument instrument) {
        this.instrument = instrument;
    }
}

File: Frontend/src/app/Class/ArtistInstrument.ts
import { Artist } from "./Artist";
import { Instrument } from "./Instrument";

export class ArtistInstrument {
  id: number;
  artist: Artist;
  instrument: Instrument;

  constructor(obj?: any) {
    this.id = (obj && obj.id) || 0;
    this.artist = (obj && obj.artist) || null;
    this.instrument = (obj && obj.instrument) || null;
  }

  // Getters

  getId(): number {
    return this.id;
  }

  getArtist(): Artist {
    return this.artist;
  }

  getInstrument(): Instrument {
    return this.instrument;
  }

  // Setters

  setArtist(artist: Artist) {
    this.artist = artist;
  }

  setInstrument(instrument: Instrument) {
    this.instrument = instrument;
  }
}
